### ISSUE TITLE
Collaboration with Node.js Community Committee / Working Group Members

### ISSUE THREAD

**Issue by WaleedAshraf** [nodeschool/organizers#520]
Hi Everyone, I'm an organizer for NodeSchool/Lahore chapter and ~observer~ member in [Node.js/community-commitee](https://github.com/nodejs/community-committee)

There has been a discussion going on, to create Node.js Community Chapters or to have a collaboration with NodeSchool.
Related Issue: [Initiative for Nodejs Community Chapters](https://github.com/nodejs/community-committee/issues/280)

In the last meeting of community-committee, we discussed creating a public repo inside [Node.js Org](https://github.com/nodejs) for community events. This would be a place where anyone can create related issues and seek help from Node.js working group members.
It will also include a team of people from Node.js working groups who can help with events/workshops related issues.

It will open a channel of communication for NodeSchool Organizers with Node.js working group members.

I need feedback/suggestion from NodeSchool Organizers on this.

Thanks,

**Comment by WaleedAshraf** [comment]
Ping @nodeschool/chapter-organizers @nodeschool/core

**Comment by oe** [comment]
👋 Berlin chapter organizer and Node.js Community Committee member here. It'd mean a lot to the Node.js project if we could get some sort of collaboration going between the two sides. I understand that most people here do this in their free time, but even just dropping a few ideas or feedback on the proposed structure would help a ton. Thank you all!

**Comment by Sequoia** [comment]
Hi All! I've been authorized (by myself) to offer _one nodeschool organizer's opinion™_ and it is as follows:

Nodeschool chapters offer only whatever you bring to them, which is to say a nodeschool chapter is whatever you make it. With this in mind, I'd suggest that there is no guideline as to whether to start a nodeschool chapter or Community Chapter–it's strictly up to your preference!

Some possible differences:

* Nodeschool is 100% volunteer at the organization level.
* Nodeschool is a **federation**, meaning there's no actual central committee or governing body, strictly speaking. This means there are few guidelines about how you run your chapter or events, for example you can choose to solicit sponsorship or not. The main guiding principal (imo) is "make events free to attendees."
* Nodeschool has no corporate sponsorship or affiliation at the organization level (node community may?).
* Nodeschool has been around longer than the Node foundation. While very active in years past, participation may have petered off (hard to say without checking–see the "we're a federation/no central governing body" point). **Node.js Community Chapters _may_ be a more active project at this point in time.**

That's about all I know! If you would like to start a chapter, let us know and we can set up a team etc. for you. 👍

**Comment by Sequoia** [comment]
@oe Is there specific feedback or input you're seeking? Speaking (again) for myself, I'm happy to lend a hand to the ol' node community when I am able, but your query as it stands is too open ended to be actionable. Can you specify?

In any event I'm glad to hear there's community organizing going forward!

**Comment by finnp** [comment]
Another Berlin chapter organizer here 👋 That sounds like a pretty good start.

**Comment by jpwesselink** [comment]
@oe I'm on the same page as @Sequoia here: if you can be more specific then I'd gladly be of service.

**Comment by martinheidegger** [comment]
As far as I can tell Node.js Community Chapters are - more or less - properly organized, local groups of people with support from the Node Foundation whereas NodeSchool chapters are just more or less organized, local groups. How are local groups - that organize NodeSchool events - able to receive recognition/support from the Node Foundation? Would it be okay to push a label on the homepage/group that shows the support of the Node for a local group?

**Comment by codeekage** [comment]
We currently don't have something we can label Node.js Community Chapters, and that's what we are working on if we can work with the NodeSchools and NodeSchool's Organizers to start up Node.js Community Chapters.

**Comment by martinheidegger** [comment]
My suggestion:
1. Node.js (foundation = NF) sets up a chapter program (how people can form chapters).
2. NF makes labels that link to a page where chapters can be "verified"
3. NF offers NodeSchool chapters to get NF recognition as a chapter
4. NodeSchool chapters put the NF label on the NodeSchool list of chapters (once they are a NF community chapter).

If a NodeSchool chapter doesn't want to become a NF chapter, is not maintained anymore or is not compatible with NF it can just stay as-is.

**Comment by codeekage** [comment]
Awesome @martinheidegger that will be highly considered.

**Comment by huberv** [comment]
Hi all, I'm the initiator and main organizer of the NodeSchool Passau chapter.
I totally back the proposal of @martinheidegger !

**Comment by WaleedAshraf** [comment]
NCWG = Nodejs Community Working Groups
NS = NodeSchool

@Sequoia thanks for the details on NodeSchool 👍. But I'm still confused in if you liked the idea of having a communication channel between NCWG and NS chapter? And to your points, AFAIK NCWG (like Community Committee) are also 100% volunteer. And **YES**, with this collaboration between NS and NCWG, we don't want to put corporate sponsorship or affiliation at the organization level. NS is an independent federation and will/should remain independent.

I believe this can help both. Like, NCWG are running a mentorship program where dozens of mentors and mentees have signed up. A NS chapter can ask for mentors if they volunteer to conduct a NodeSchool workshop? Similarly, we (NS organizers) can inform mentees about NS events.
There are dozens of other opportunities of collaboration between both.

martinheidegger is right. Maybe NS is more/less local groups and NCWG are more/less properly organized groups to support Nodejs Foundation. But we all have a common goal, that is to help grow Node.js Community and Node.js ecosystem, make it stronger in every possible way, it could be help training (on development) others through workshops/events/talks OR through a mentorship program.

@martinheidegger great suggestion.
What I think we can do is:

- Create repo "Community Chapters" inside Node.js
- Add details on
  - What is "Community Chapters" i.e NodeSchool
  - What community chapter does? (Could be workshops/events/talks anything)
  - Describe NodeSchool (what it is, how it works)
  - How to create a community chapter? Link to NodeSchool docs.
  - Mention that "All NodeSchool chapters are Node.js Community Chapters"
  - Add a list of all NodeSchool Chapter with the link to their site.

- We can create a label/logo "<_Nodejs Logo_> | Community Chapter"
- NodeSchool chapters may add that logo to their site/repo with link "Nodejs/Community Chpaters"

**Comment by codeekage** [comment]
@WaleedAshraf mentioning all NodeSchool chapters are Node.js Community Chapter.... mmmmmmmm? is almost like taking away the federation from NodeSchool. 

What I think is a NodeSchool can either decide to affiliate with a Node.js Community Chapter or not. 
Saying NodeSchool is now a Node.js Community Chapter is really confusing...........

**Comment by WaleedAshraf** [comment]
@codeekage right,
Let NS chapter decide what they want.

I believe all NS chapters are already awesome. they have COC, contribution guides, and everything. Becoming Node.js  Community Chapter would be just adding NS chapter name in Node.js/CommunityChpater repo and adding a label on NS chapter. 

Maybe Node.js/CommunityChpater also add some additional points on COC and contribution. that's it.

**Comment by jarrettch** [comment]
I'm one of the organizers with the NodeSchool in Los Angeles. We'd love to be featured on any such Community Chapter page and be given whatever "official" status that entails. Interested to learn more about what a Node.js/Community Chapter would entail outside of the NodeSchool part, seems like there will be more discussion around that?

**Comment by amitsin6h** [comment]
Hello all I am organizer from **Nodeschool Jharkhand** . Thats really a great idea and we would love to featured on any such community chapter, as what @jarrettch said :tada: :tada: :tada:

**Comment by gartox** [comment]
Hello folks, I am one of organizers from **Nodeschool Tabasco**, this idea sounds very good, We like to be part of this, how  can we collaborate @WaleedAshraf ?

**Comment by WaleedAshraf** [comment]
@gartox no update yet. I'll ping some members from Nodejs Community to get this finalized and we also need a final review from NodeSchool Organizers / Core Team.

**Comment by a0viedo** [comment]
I think what other people intended to say describing the situation is that you won't get someone's approval because there are no governors. The team "NodeSchool Organizers", from my point of view, are _trusted_ members but their opinions won't represent the desire or intent of the majority.

Basically what I'm saying is that NodeSchool members have guidelines but other than that, they can collaborate (as group or individuals) with any project they please.

**Comment by WaleedAshraf** [comment]
@a0viedo I agree. But I still need to know if enough people are interested in this because it requires some work to be done at Node.js Org side (like creating a repo).

+ If it goes live, I want to notify all NodeSchool chapters about this change so that can decide if they want to have this collaboration or not. Maybe we can even put it somewhere on the NodeSchool main website?

**Comment by gartox** [comment]
Thanks @WaleedAshraf, keep us aware of any update.

**Comment by WaleedAshraf** [comment]
Hi everyone,

In last Node.js Community Committee meeting, we discussed about the possibility of collaboration with communities like NodeSchool etc. We are also working on [documenting partner communities](https://github.com/nodejs/community-committee/issues/326). Though nothing is final yet.

As a NodeSchool organizer, I need to make a list of benefits that NodeSchool chapter may get if there is a collaboration between a **NodeSchool Chapter** and **Node.js  Foundation Community Committee / Working Group**.

Here are few benefits I think of:
- Oppertunity to collabortate with more Node.js / Javascript developers and community members.
- Chapter Organizers can easily take input/feedback from Node.js Foundation members/collaborators.
- Chapter can easily invite people from Node.js Foundation as speaker if needed.
- Easily advertise/publicity of NodeSchool chapter events/workshops thourgh Node.js Foundation. This can be done through opening an issue in relevant Node.js Foundation repo for discussion or through social media etc.
- Chapter can add Node.js logo on its website and add text like "in collaboration with Node.js Foundation."
- Chapter can easily request for Node.js swag for events or permission to print it by self.
- Chapter can take feedback from its event/workshop attendess and communicate it to Node.js Foundation members easily.

I need feedback from everyone. Kindly comment if you agree with these benefits and feel free to share more.
 
Ping @nodeschool/chapter-organizers

**Comment by Sequoia** [comment]
## Proposal

We put a link on nodeschool.io or maybe just on the readme for this project ( nodeschool/organizers ) saying "Want to affiliate with the Node.js Foundation? Click here for more info." and it links to a page on the Node foundation site.

@WaleedAshraf: this would allow people to find their way from Nodeschool to the Node Foundation initiative, and that landing page (of your creation) could have whatever pitch you want to make describing benefits etc..

Goals achieved:
1. People have a way to find the Node Foundation CCWG & learn the benefits
2. We don't need to try to get buy in from Nodeschool "as a whole" (currently impossible given the federated structure of nodeschool).

I'll stand by for :-1:/:+1:s!

**Comment by ProfNandaa** [comment]
LGTM 👍

**Comment by WaleedAshraf** [comment]
@Sequoia Totally agree with your approach. 👍 

@everyone I'll share these benefits in next Node.js Community Committee meeting (in next few days).
If anyone wants to add anything, please do it now.

**Comment by nebrius** [comment]
Hi folks, one of the San Francisco organizers here. I chatted with @amiller-gh at Node+JS Interactive about this, and we're down to help pilot this initiative.

cc @Xabadu @tlhunter

**Comment by nebrius** [comment]
As far as the benefits go, here's my thoughts on it. I want to preface my comments though by saying that San Francisco is, as always, a bit unique when it comes to events and most likely do not represent the position of other chapters.

**Opportunity to collaborate with more Node.js / JavaScript developers and community members.**

This is indeed a great benefit, but I'm skeptical how this program will achieve that. The Node.js collaborators in the SF Bay Area have been involved with NodeSchool SF for some time (yay, you folks are awesome!).

I am a little more interested in how the Node.js Foundation can help us expand our audience reach. Do you have any concrete ideas how you would go about this? (Listing on the website alone probably won't help much IMO).

**Chapter Organizers can easily take input/feedback from Node.js Foundation members/collaborators.**

Can you explain how this will benefit us? I can see how this benefits the Node.js Foundation, but I'm afraid I'm missing how it will help us achieve our goals.

**Chapter can easily invite people from Node.js Foundation as speaker if needed.**

This may be unique to us, but we don't have speakers and that format of education doesn't align with how we like to run events.

**Easily advertise/publicity of NodeSchool chapter events/workshops thourgh Node.js Foundation. This can be done through opening an issue in relevant Node.js Foundation repo for discussion or through social media etc.**

I can see this being valuable if the process is quick and efficient, but it needs to be low-investment to work. My concern here is that it's One More Thing To Do, and since most of us volunteer in our free time this can be an issue of commitment.

**Chapter can add Node.js logo on its website and add text like "in collaboration with Node.js Foundation."**

I gotta admit, I'm really "meh" about this. Generally speaking, the SF community doesn't really pay attention to logos and sponsorship because we're constantly inundated with it. I could be wrong, but my hunch is that this wouldn't have an impact on growing our audience.

**Chapter can easily request for Node.js swag for events or permission to print it by self.**

Similar to an above comment, this could be cool _if_ it's really low time investment for us. It'd be nice to have, but it's not something I'd go out of my way for.

**Chapter can take feedback from its event/workshop attendees and communicate it to Node.js Foundation members easily.**

Similar to an above comment, this sounds like it'd be valuable for the Node.js Foundation, but how does it benefit us? (I think this should go in the "benefits for CommComm" column myself, but maybe I'm missing something).

**Comment by D3Portillo** [comment]
Hi San Miguel team organizer here, well we indeed are a small team not more than 14 members that love doing coding stuff, I see a great opportunity to be part/affiliate with node js thus we could "due to location" make a live speech with them, I hope you guys allow this, could be great having some swag, so that people get even more excited :fire: :fire: :+1:

**Comment by Sequoia** [comment]
@WaleedAshraf It looks like my proposal [above](https://github.com/nodeschool/organizers/issues/520#issuecomment-413022495) was popular, so let's go with that. Can you submit a PR that updates the readme with the link to your org? TIA!

**Comment by WaleedAshraf** [comment]
thanks @nebrius for sharing your thoughts.

I'll try explain bit more about points you raised but before that, I want to clear that,
- I got involved with NodeSchool around 4,5 months ago, and Node.js Foundation around 8,9 months ago. Not a person who has spent a lot of time with any of these Orgs and I know there are people on both sides who have been around for years (I think you also?). Those are the people I want input from, they can suggest this in a much better way than me.

- My understanding as a whole is it's not just about listing down benefits in bullet points to take this forward. (well, yes! that should also be done). But there is one main, bigger reason, which should be enough to push this forward. All NodeSchool chapters are putting time & effort in community education/code+learn around Javascript/OpenSource/Nodejs and much more. And so, Node.js Community is also running initiatives like [mentorship ](https://github.com/nodejs/mentorship) and [user-feedback](https://github.com/nodejs/user-feedback) which are directly linked with community and people interested in learning more. And hopefully, there will be more like these in the future. A bridge between both can open many opportunities for both in terms of better communication and effective efforts.

- To add more, with JS Foundation and Node.js Foundation merge happening soon, this collaboration can help reach a lot bigger audience on both sides then before.

- Points I listed are just possibilities and they can vary from one chapter to other chapters. Maybe for SF chapter speakers is not a big issue, but for others, It can be. Same goes for all other points.

> but it's not something I'd go out of my way for.

I'm not sure how difficult this could be for a chapter? As far as I understand, this is the easiest collaboration any chapter can have without putting any extra time/effort.

> this should go in the "benefits for CommComm"

I also consider that these benefits are on both sides. Not just CommComm or NodeSchool chapter. I should update my previous comment.

> logo on its website 

Yup, I agree. This would be a difficult task from Foundation side also because of legal and IP stuff. But still, it may happen.

> Chapter Organizers can easily take input/feedback Foundation members.

I see this as there is one person who is involved with JavaScript/Node.js development as being part of Foundation. and they are also running a NodeSchool chapter to educate/help community about these technologies. They can guide community in much different way. For example, about upcoming features/roadmap in JavaScript/Node.js plateform. (which are already open for public but It's all about communication.)

I'm looking forward for more suggestions from you and others to make it as much better as we can. 👍
